lighthouse-viewer/app/src/viewer-ui-features.js

```diff
@@ -18,12 +19,19 @@ import {ReportUIFeatures} from '../../../report/renderer/report-ui-features.js';
 export class ViewerUIFeatures extends ReportUIFeatures {
   /**
    * @param {DOM} dom
-   * @param {?function(LH.Result): void} saveGistCallback
+   * @param {{saveGist?: function(LH.Result): void, refresh: function(LH.Result): void}} callbacks
    */
-  constructor(dom, saveGistCallback) {
+  constructor(dom, callbacks) {
     super(dom);
 
-    this._saveGistCallback = saveGistCallback;
+    this._callbacks = callbacks;
+    this._swapLocales = new SwapLocaleFeature(this, this._dom, {
+      async fetchData(localeModuleName) {
+        const response = await fetch(`./locales/${localeModuleName}.json`);
+        return response.json();
+      },
+      refresh: callbacks.refresh,
```

Comment:
nit: this might be a recipe for a future subtle bug (`this._callbacks.refresh` is there but swap locale features is bound to whatever was in the constructor)

preference would be to revert the saving of `_callbacks` and only the save the one we really need, or use a delegate here so that `_callbacks.refresh` is used

Reply:
Not 100% following, so I'll try to restate and you can let me know if I got it right.

The provided callback for this refresh function is an arrow function, so it is scoped lexically. Are you saying the subtle bug is if `refresh` in LighthouseReportViewer is defined as a normal function without binding to the correct `this`, then the code will error when called in `SwapLocaleFeature`?

Reply:
> Are you saying the subtle bug is if refresh in LighthouseReportViewer is defined as a normal function without binding to the correct this, then the code will error when called in SwapLocaleFeature?

No, I'm saying if we ever have a function `setCallbacks(callbacks) { this._callbacks = callbacks }` it will update the saveGist function but not do anything to the refresh one, i.e. there's no real point to saving `callbacks.refresh` to `this` at the moment either because it's never used. 

Not saying we're likely to do that (that's why it's a nit), but having a `this._callbacks` that contains `refresh` feels a smidge misleading, so preference would be to either only save `this._saveGistCallback` like the code was before and leave `refresh: callbacks.refresh` as it is OR change to `refresh: lhr => this._callbacks.refresh(lhr)`.

Reply:
You could also do nothing, it is a nit afterall and fairly unlikely to happen (or at the very least require other removeEventListener/addEventListener work that points this out).

Reply:
It might be helpful to split up the functionality a bit. It does feel a little weird for `swap-locale-feature` to be reaching into `reportUiFeatures.json` and doing the locale swapping itself.

One option is that `lighthouse-report-viewer` takes on the loading of locale files, swapping them, and keeps json replacement internal, and `swap-locale-feature` is limited more to a menu that takes a current locale and generates a `<select>` from `format.getAvailableLocales`, the calls a callback when a locale is selected.